src/popup/js/RvH.js

```diff
@@ -43,24 +43,33 @@ $(function() {
 	/**
 	 * Populate the popup
 	 */
-	function populate(data) {
-
-		// robots.txt
-		var robots = data.robots;
-		if (robots !== false) {
-			$('#robots').html('<pre>' + RvH.common.Util.htmlEntities(robots) + '</pre>');
-		}
-		else {
-			$('#robots').html('<p>' + chrome.i18n.getMessage("fileNotFound", ["robots.txt"]) + '</p>');	
-		}
+	 function populate(data) {
+
+ 		// robots.txt
+ 		var robots = data.robots;
+ 		chrome.tabs.query({
+ 				'active': true,
+ 				'windowId': chrome.windows.WINDOW_ID_CURRENT
+ 		}, function(tabs) {
+ 				var tablink = tabs[0].url;
+ 				//Regex to match only the protocol and host of a url
+ 				tablink = tablink.match(/^[\w-]+:\/{2,}\[?([\w\.:-]+)\]?(?::[0-9]*)?/)[0];
+ 				if (robots !== false) {
+ 						var htmlEntities = RvH.common.Util.htmlEntities(robots);
```

Comment:
There's a lot of repeating going on in these files to get the host and then convert html entities and add links. This could all be converted to a common util function and then used for both the robots and humans text display e.g.

```javascript
RvH.common.Util.parseText(robots, 'robots.txt', function(text) {
  $('#robots').html(text);
});

...

// common/rvh/Util.js

parseText(data, file, callback) {
  if (data === false) {
    callback('<div class="alert alert-danger">' + chrome.i18n.getMessage("fileNotFound", [file]) + '</div>');
  }
  else {
    chrome.tabs.query({
      'active': true,
      'windowId': chrome.windows.WINDOW_ID_CURRENT
    }, function(tabs) {
      var tablink = tabs[0].url.match(/^[\w-]+:\/{2,}\[?([\w\.:-]+)\]?(?::[0-9]*)?/)[0];
      var parsedData = this.addLinks( // <-- rename the parseAsLink method
        this.htmlEntities(data)
      );
      callback('<pre>' + parsedData + '</pre>');
    });
  }
}
```